ddtrace/tracer/option.go

```diff
@@ -52,6 +55,15 @@ func WithDebugMode(enabled bool) StartOption {
 	}
 }
 
+// NoErrorStack will disable the automatic generation of error stacks when setting the
+// error tag with a value of type error. However, it will still be possible to attach the
+// error stack manually, if desired, using the "error.stack" tag.
+func NoErrorStack() StartOption {
+	return func(c *config) {
+		c.noErrorStack = true
+	}
+}
```

Comment:
Can we have a callback that generates call stack?
A callback that carries information about the span (same as `span` struct but public).
This way we can also optimise the stack trace generation for certain cases.

Reply:
Spans were intentionally designed as write-only. Perhaps this was not the best decision, we will tell in time, but for now, in v1 we don't plan on changing this.

As for having a callback, this is something we definitely avoided because introducing user-definable callbacks throughout the pipeline would just increase the the chances of users "shooting themselves" in the foot by seriously wearing down performance. Better stay safe.

This should do for now. It will disable automatic stack traces. Wherever you still want a stack trace, you may do so automatically by calling:
```
span.SetTag(ext.ErrorStack, debug.Stack())
```
No need for a callback IMO.

For more custom cases, there is always the option of manual instrumentation.